sklearn/manifold/t_sne.py

```diff
@@ -423,18 +423,19 @@ def trustworthiness(X, X_embedded, n_neighbors=5, precomputed=False):
         dist_X = X
     else:
         dist_X = pairwise_distances(X, squared=True)
-    ind_X = np.argsort(dist_X, axis=1)
+
+    ind_X = np.argsort(dist_X, axis=1)[:, 1:]
```

Comment:
Does this `1:` slice still work if there are duplicate points (i.e. multiple pairs have dist=0)?

Reply:
Maybe we should use a NearestNeighbors for this as well?

Reply:
Good catch @jnothman! Here is a snippet that shows that there is a problem indeed:

```py
import numpy as np

from sklearn.metrics import pairwise_distances

rng = np.random.RandomState(0)
arr = rng.randn(1000, 10)
arr_with_duplicates = np.vstack((arr, arr))
dist = pairwise_distances(arr_with_duplicates)
print((np.argsort(dist, axis=1)[:, 0] ==
       np.arange(arr_with_duplicates.shape[0])).all())
```

It outputs `False`.